File: hw/gualotuna/u3/hw30/Strategy/src/main/java/model/InsertionSort.java
package model;

import controller.SortingStrategy;

/**
 *
 * @author  Brayan Gualotuña  
 */
public class InsertionSort implements SortingStrategy {

    @Override
    public int[] sort(int[] array) {
        int n = array.length;
        for (int i = 1; i < n; i++) {
            int key = array[i];
            int j = i - 1;

            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j = j - 1;
            }
            array[j + 1] = key;
        }
        return array;
    }

}

File: exams/rodriguez/ExamU3/Strategy Pattern/Strategy Pattern/model/insertion_sort.py
from model.sorting_strategy import SortingStrategy

class InsertionSort(SortingStrategy):
    def __init__(self):
        print("Creating InsertionSort strategy")

    def sort(self, data):
        for i in range(1, len(data)):
            key = data[i]
            j = i-1
            while j >= 0 and key < data[j]:
                data[j+1] = data[j]
                j -= 1
            data[j+1] = key
        return data